### PR TITLE
Add Nginx stub status to metricbeat

### PR DESCRIPTION
**Pull request by mrkschan**

### CODE REVIEW

File: metricbeat/docs/modules/nginx.asciidoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+[[metricbeat-module-nginx]]
+== Nginx Module
+
+This module periodically fetches metrics from https://nginx.org/[Nginx] servers.
+
+
+
+=== Example Configuration
+
+The Apache module supports the standard configuration options which can be found
```
Comment (andrewkroh) [inline metricbeat/docs/modules/nginx.asciidoc]: `s/Apache/Nginx/`

File: metricbeat/module/nginx/stubstatus/stubstatus.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	// the host config.
+	defaultScheme = "http"
+
+	// defaultPath is the default path to the mod_status endpoint on the
+	// Apache HTTPD server.
+	defaultPath = "/server-status"
+)
+
+var (
+	debugf = logp.MakeDebug("apache-status")
```
Comment (andrewkroh) [inline metricbeat/module/nginx/stubstatus/stubstatus.go]: `s/apache-status/nginx-stubstatus/` or maybe just `nginx`. Not sure how much granularity we need.

File: metricbeat/module/nginx/stubstatus/stubstatus.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+const (
+	// defaultScheme is the default scheme to use when it is not specified in
+	// the host config.
+	defaultScheme = "http"
+
+	// defaultPath is the default path to the mod_status endpoint on the
```
Comment (andrewkroh) [inline metricbeat/module/nginx/stubstatus/stubstatus.go]: Comment needs updated for nginx.

File: metricbeat/module/nginx/stubstatus/stubstatus.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	if u.Scheme == "" {
+		// Add scheme and re-parse.
+		u, err = url.Parse(fmt.Sprintf("%s://%s", "http", rawHost))
+		if err != nil {
+			return nil, fmt.Errorf("error parsing nginx host: %v", err)
+		}
+	}
+
+	if u.Host == "" {
+		return nil, fmt.Errorf("error parsing apache host: empty host")
```
Comment (andrewkroh) [inline metricbeat/module/nginx/stubstatus/stubstatus.go]: `s/apache/nginx/`

File: metricbeat/module/nginx/stubstatus/data.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		writing  int
+		waiting  int
+	)
+
+	var re *regexp.Regexp
+	scanner := bufio.NewScanner(body)
+
+	// Parse active connections.
+	scanner.Scan()
+	re = regexp.MustCompile("Active connections: (\\d+)")
```
Comment (andrewkroh) [inline metricbeat/module/nginx/stubstatus/data.go]: I recommend declaring all the regular expressions as package level variables. Currently this is rebuilding the regular expression each time it parses. It also defers the validation of the regular expression until the method is invoked.

So if you move all the `regexp.MustCompile` calls into a `var (   )` block, then the regular expressions are only compiled once (performance boost) and they are validated at start-up irregardless of whether the module is used.

**Comment by andrewkroh** [comment]
Looks good, I just had a few minor comments. Thanks for your contribution!

**Comment by ruflin** [comment]
@mrkschan Thanks a lot for the contribution and already integrating our most recent metricbeat changes. I'm happy to merge it as it is and do the changes found by @andrewkroh afterwards so we can help with it, or you can push them yourself. What do you prefer?

**Comment by mrkschan** [comment]
fixed, please let me do if there's any other issue :)

**Comment by ruflin** [comment]
@mrkschan All good. Thanks a lot 👍

**Pull request merged**